### PR TITLE
refactor(dropdown): fix click outside behavior

### PR DESCRIPTION
**Pull request by masoudmanson**
## Summary

**Dropdown**
GitHub issue: #414 

Amend the `Dropdown` menu to stay open only for the button variant.

### CODE REVIEW

File: src/core/Dropdown/index.tsx [hunk 1]
```diff
@@ -57,7 +67,7 @@ export interface DropdownProps<Multiple> {
 }
 
 // eslint-disable-next-line sonarjs/cognitive-complexity
-const Dropdown = <Multiple extends boolean | undefined = false>({
+const Dropdown = <Multiple extends boolean | undefined = undefined>({
```
Comment (tihuan) [inline src/core/Dropdown/index.tsx:70]: hmm line 73 is `multiple = false`, so I'm thinking the type needs to be `Multiple extends boolean | undefined = false` too? Unless I missed something 😆 

Thank you!

File: src/core/Dropdown/index.tsx [hunk 2]
```diff
@@ -184,7 +194,9 @@ const Dropdown = <Multiple extends boolean | undefined = false>({
 
   function handleClickAway() {
     if (open) {
-      setOpen(false);
+      if (closeOnBlur && !shouldShowButtons) {
```
Comment (tihuan) [inline src/core/Dropdown/index.tsx:197]: I wonder if some comments would help in this file and functions to explain why we have certain conditions and diverged behavior, so we give readers enough context and information that will help them understand the code and make modifications down the road more confidently?

**Review by tihuan — COMMENTED** [review]
Thanks for refactoring the code, @masoudmanson !

Do you know why there's no Chromatic deployment in PR test section 🤔 ?

Was thinking that @clarsen-czi  could help test this out in Chromatic 😄 

Thank you!

**Comment by masoudmanson** [comment]
> Details

Hmm, I can see a Chromatic deployment in PR test sections :/ 
Here is the link for the deployed storybook instance:

https://616981d9028812003ade73f3-yvagcriyjb.chromatic.com

File: src/core/Dropdown/index.tsx (hunk 1 above)
Comment (masoudmanson) [reply src/core/Dropdown/index.tsx:70]: Aah, that makes sense. I'll fix it. 🥸

File: src/core/Dropdown/index.tsx (hunk 2 above)
Comment (masoudmanson) [reply src/core/Dropdown/index.tsx:197]: Sure, I'll add some comments. 🤓

**Review by clarsen-czi — APPROVED** [review]

**Comment by tihuan** [comment]
> > Details
> 
> Hmm, I can see a Chromatic deployment in PR test sections :/ Here is the link for the deployed storybook instance:
> 
> https://616981d9028812003ade73f3-yvagcriyjb.chromatic.com

Ooh I see the chromatic deployment in test now too. Thank you! Not sure why they were missing for me -.- Sorry for the false alarm 🚨 😆

**Pull request merged**